tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1920,8 +1928,12 @@ static llvm::Value* GetStartOffsetX(const KernelMappingScheme& mapping_scheme,
                                     llvm::Value* thread_id_x,
                                     llvm::Type* index_ty,
                                     llvm::IRBuilder<>* b) {
-  if (mapping_scheme.DilatedX()) {
+  if (mapping_scheme.GetIndexingOrder() == kStridedIndexingX) {
     return thread_id_x;
+  } else if (mapping_scheme.GetIndexingOrder() == kLinearStridedIndexingX) {
+    int vector_size = mapping_scheme.GetVectorSize();
+    return b->CreateMul(thread_id_x,
+                        llvm::ConstantInt::get(index_ty, vector_size));
   }
   int64 x_num_steps =
```

Comment:
Perhaps insert a CHECK on what the indexing order has to be in the third case?

Reply:
I mean, you are assuming that the indexing order has to be linear, so it's useful to check for that.